synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -956,6 +993,13 @@ void Widget_Timetrack::on_waypoint_action_changed()
 	signal_action_state_changed().emit();
 }
 
+void Widget_Timetrack::on_waypoint_selection_changed()
+{
+	std::vector<WaypointItem*> selection = waypoint_sd.get_selected_items();
+	bool show_tool_bar_buttons = (selection.size() != 0);
+	signal_update_interpolation_buttons_visiblity().emit(show_tool_bar_buttons);
```

Comment:
Instead of this signal name, that describe an action other entity should perform, the name should describe what event happened. Example: waypoint_selection_changed

Reply:
oh yes I went a little unorthodox with this one i guess xd, done